app/frontend/components/FeatureGrid.tsx

```diff
@@ -1,28 +1,21 @@
-import { useListFeatures } from "../lib/api-client";
 import { FeatureCard } from "./FeatureCard";
+import { Feature } from "../lib/api-client.schemas";
 
-export const FeatureGrid = () => {
-  const { data, error, isPending } = useListFeatures({
-    projectionExpression: "SK,title,imagePath",
-  });
-
-  if (isPending) {
-    return <h1 className="text-3xl flex justify-center">Loading...</h1>;
-  }
-
-  if (error) {
-    return <h1 className="text-3xl flex justify-center">Error: {error.message}</h1>;
-  }
-
-  const { features } = data.data;
+type FeatureGridProps = {
+  features: Feature[];
+  searchTerm: string;
+};
 
+export const FeatureGrid = ({ features, searchTerm }: FeatureGridProps) => {
   return (
     <section className="w-[80%] mx-auto p-10 grid grid-cols-1 lg:grid-cols-2 xl:grid-cols-3 2xl:grid-cols-4 gap-6 place-items-center">
-      {features?.map(({ imagePath, SK, title }) => {
-        if (!imagePath || !SK || !title) throw new Error("There was an error retreiving or rendering feature data.");
-        const id = SK.split("#")[1];
-        return <FeatureCard key={id} imgSrc={imagePath} title={title} id={id} />;
-      })}
+      {features
+        .filter(({ title }) => title?.toLowerCase().includes(searchTerm.toLowerCase()))
+        .map(({ imagePath, SK, title }) => {
+          if (!imagePath || !SK || !title) throw new Error("There was an error retreiving or rendering feature data.");
+          const id = SK.split("#")[1]; // Ex: "FEATURE#123" -> "123"
```

Comment:
i saw there was a backend function for extracting the feature id only should it not be its own function in frontend as well unless are we sure THIS is the only place it will be used? hahaha

Reply:
Let's say `const id = SK.split("#")[1];` gets repeated three or four times and I'll make a function for it, but I'd rather not introduce indirection for a one liner

Reply:
understood, so far this is the only place i saw it in frontend for now that does work.